### DISCUSSION TITLE
issue: react-select with Controlled don't work with setValue

### DISCUSSION THREAD

**Discussion by sunyeongchoi** [react-hook-form/react-hook-form#7760] [Q&A] [answered]
`react-select` with `Controlled` don't work with [setValue](https://react-hook-form.com/api/useform/setvalue) method

If I check the ref value using the watch() method, the value is displayed as intended, but the external value of the selectbox has not changed.

What should I do when I want to change the external value of the selectbox using setValue as well?

is this an error? Or is the usage wrong?

code sandbox: https://codesandbox.io/s/react-hook-form-controller-react-select-uwpx1?file=/src/App.js

**Comment by Moshyfawn** [comment, marked as answer]
From what I understand, you're passing `undefined` into your select value initially, because your `defaultValue` is a `""` _(empty string)_, so
```js
value={options.find((c) => c.value === value)}
```
results in `undefined`, which means that your select input is uncontrolled, aka the source of truth for your select value is the input, not RHF.
Use [Nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator) to default to an empty string to init your select input as controlled
```js
value={options.find((c) => c.value === value) ?? ""}
```
See [this CSB](https://codesandbox.io/s/react-hook-form-controller-react-select-forked-39gvm?file=/src/App.js) for a working example.

As a side note, there's [a `resetField` form method](https://react-hook-form.com/api/useform/resetfield/#main) that's more suitable to clear the inputs like you want to do with your select

**Reply by bluebill1049** [reply]
react select default state should be an object.

https://codesandbox.io/s/react-hook-form-controller-react-select-forked-tl0g5?file=/src/App.js

```tsx
import "./styles.css";
import { useForm, Controller } from "react-hook-form";
import Select from "react-select";

export default function App() {
  const { control, setValue } = useForm();
  const handleReset = () => {
    setValue("reset", { value: "", label: "Select..." });
  };

  const options = [
    { label: "Select...", value: "" },
    { label: "test", value: "test" },
    { label: "test2", value: "test2" },
    { label: "test3", value: "test3" }
  ];

  return (
    <>
      <label htmlFor="reset">setValue Test</label>
      <Controller
        defaultValue={{ value: "", label: "Select..." }}
        control={control}
        name="reset"
        render={({ field: { value, onBlur, onChange, ref } }) => (
          <Select
            ref={ref}
            value={value}
            options={options}
            onChange={onChange}
            onBlur={onBlur}
          />
        )}
      />
      <button type="button" onClick={handleReset}>
        Reset
      </button>
    </>
  );
}

```

**Reply by sunyeongchoi** [reply]
Thanks I solved it !!
Thank you very much :)

**Reply by Moshyfawn** [reply]
Ah, makes sense, @bluebill1049. I was referring to the CSB provided by the OP for the defaultValue.

**Reply by bluebill1049** [reply]
👍 thanks for the detailed help @Moshyfawn

**Reply by sunyeongchoi** [reply]
@bluebill1049 
But I have a question about this code.
If defaultValue exists, the placeholder doesn't work as expected.

So should I use defaultValue instead of placeholder?

codesandbox: https://codesandbox.io/s/react-hook-form-controller-react-select-forked-1skrr?file=/src/App.js

**Reply by Moshyfawn** [reply]
Your `option.label` works as a placeholder in this case.

**Reply by Moshyfawn** [reply]
[An example CSB](https://codesandbox.io/s/react-hook-form-controller-react-select-forked-3ey58?file=/src/App.js)

**Reply by sunyeongchoi** [reply]
Oh thanks for the detailed explanation :)
